kratos/includes/global_pointer.h

```diff
@@ -196,6 +196,17 @@ class GlobalPointer {
     return mDataPointer;
   }
 
+  /**
+   * @brief Overloads the '==' operator to compare two GlobalPointer objects of the
+   * same template type. Returns true if the underlying pointers are equal.
+   * @param rOther The GlobalPointer object to be compared.
+   * @return true if the underlying pointers are equal, false otherwise.
+   */
+  bool operator==(const GlobalPointer& rOther)
+  {
+      return this->get() == rOther.get();
```

Comment:
```suggestion
#ifdef KRATOS_USING_MPI
    return this->get() == rOther.get() && this->mRank == rOther->mRank;
#else 
    return this->get() == rOther.get();
#endif
```

Reply:
Hello

Reply:
It is a on in a million chance, but if you end up running kratos in a machine with two separate memory spaces and you compare a (0,0xFFFFFFABCD123) with (1,0xFFFFFFABCD123) you are screwed.

Reply:
Yes, I realized, that why I applied your suggestion. Feel free to approve